lang/wasm.cc

```diff
@@ -218,7 +259,21 @@ struct from_val_visitor {
             return bytes_opt{};
         }
         data += sizeof(int32_t); // size of the return type was consumed
-        return t.decompose(t.deserialize(bytes_view(reinterpret_cast<int8_t*>(data), ret_size)));
+        bytes_opt ret = t.decompose(t.deserialize(bytes_view(reinterpret_cast<int8_t*>(data), ret_size)));
+        auto free_export = instance.get(store, "_scylla_free");
```

Comment:
Also, it looks like we should attempt to use RAII here and try to free memory also when we exit the current scope via an exception. I suppose we could create a smart-pointer-like object during "malloc" and pass it to here, where it can finally get destructed and free everything. Otherwise, we may end up leaking memory if somebody uploaded wasm code with correct allocation routines, but with a subtle bug somewhere else, e.g. during function execution. I think that right now we could end up in a situation when something was already allocated via guest malloc, but then it would never be freed because calling the actual wasm function failed and threw an exception back to the user.

Reply:
After deciding that we free arguments in wasm, we can't really use RAII here, but good catch with the _scylla_free and _scylla_malloc